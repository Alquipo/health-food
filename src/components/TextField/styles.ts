import styled, { css } from 'styled-components'
import { TextFieldProps } from '.'
import InputMask from 'react-input-mask'

export const Wrapper = styled.div<TextFieldProps>`
  ${({ theme, gridArea }) => css`
    width: 100%;

    display: flex;
    flex-direction: column;
    justify-content: center;
    grid-area: ${gridArea};
  `}
`

export const Label = styled.label`
  ${({ theme }) => css`
    display: block;
    font-size: ${theme.font.sizes.medium};
    margin-bottom: ${theme.spacings.xxsmall};
  `}
`

export const Input = styled(InputMask)`
  ${({ theme }) => css`
    display: block;
    max-width: 25rem;
    width: 100%;
    padding: ${theme.spacings.xsmall};
    border: solid 0.1rem ${theme.colors.secondary};
    border-radius: ${theme.border.radius};
    font-size: ${theme.font.sizes.medium};
    transition: ${theme.transition.fast};
    font-size: ${theme.font.sizes.small};

    &:focus,
    &:hover {
      outline: none;
    }
  `}
`

export const Error = styled.span`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.red};
  `}
`
