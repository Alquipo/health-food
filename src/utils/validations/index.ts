import * as yup from 'yup'

const nascimentoRegex =
  /^((((19|[2-9]\d)\d{2})-(0[13578]|1[02])-(0[1-9]|[12]\d|3[01]))|(((19|[2-9]\d)\d{2})-(0[13456789]|1[012])-(0[1-9]|[12]\d|30))|(((19|[2-9]\d)\d{2})-02-(0[1-9]|1\d|2[0-8]))|(((1[6-9]|[2-9]\d)(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00))-02-29))$/g

const cpfRegex = /^\d{3}.?\d{3}.?\d{3}-?\d{2}$/

const cepRegex = /^([\d]{2})\.*([\d]{3})-*([\d]{3})/

const estadoRegex =
  /^(AC|AL|AP|AM|BA|CE|DF|ES|GO|MA|MT|MS|MG|PA|PB|PR|PE|PI|RJ|RN|RS|RO|RR|SC|SP|SE|TO)$/i

export const fieldsValidationSchema = yup.object({
  name: yup.string().required('Is not allowed to be empty'),
  birthDate: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(nascimentoRegex, 'must be a valid email'),
  cpf: yup
    .string()
    .max(14, 'Must be 14 characters')
    .required('Is not allowed to be empty')
    .matches(cpfRegex, 'Invalid CPF'),
  zipCode: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(cepRegex, 'Invalid ZipCode'),
  address: yup.string().required('Is not allowed to be empty'),
  district: yup.string().required('Is not allowed to be empty'),
  number: yup.number().required('Is not allowed to be empty'),
  city: yup.string().required('Is not allowed to be empty'),
  state: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(estadoRegex, 'Invalid State')
})
