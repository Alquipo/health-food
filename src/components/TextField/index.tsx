import * as S from './styles'

import { FieldHookConfig, useField } from 'formik'

export type TextFieldProps = {
  label?: string
  placeholder?: string
  onBlur?: () => Promise<void>
  gridArea?: string
  mask?: string
}

function TextField({
  label,
  placeholder,
  onBlur,
  mask,
  ...props
}: TextFieldProps & FieldHookConfig<string>) {
  const [field, meta] = useField(props)

  return (
    <S.Wrapper gridArea={field.name}>
      {label && <S.Label htmlFor={field.name}>{label}</S.Label>}

      <S.Input
        placeholder={placeholder}
        type={props.type}
        {...field}
        onBlur={onBlur}
        autoComplete="off"
        mask={mask}
      />

      {meta.error && <S.Error>{meta.error}</S.Error>}
    </S.Wrapper>
  )
}

export default TextField
