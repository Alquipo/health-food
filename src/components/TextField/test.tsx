import { render } from 'utils/test-utils'
import TextField from '.'

const fieldMock = {}
const metaMock = {}

jest.mock('formik', () => ({
  useField: jest.fn().mockImplementation(() => {
    return [fieldMock, metaMock]
  })
}))

describe('<TextField/>', () => {
  it('Renders with Label', () => {
    const { container } = render(<TextField label="test" name="Label" />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
