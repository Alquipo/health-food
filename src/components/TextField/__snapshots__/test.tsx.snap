// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField/> Renders with Label 1`] = `
.c0 {
  width: 100%;
  max-width: 25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: block;
  font-size: 1.6rem;
  margin-bottom: 0.8rem;
}

.c2 {
  display: block;
  width: 100%;
  padding: 1.6rem;
  border: solid 0.1rem #1D164D;
  border-radius: 0.6rem;
  font-size: 1.6rem;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
  font-size: 1.4rem;
}

.c2:focus,
.c2:hover {
  outline: none;
}

<div
  class="c0"
>
  <label
    aria-labelledby="test"
    class="c1"
  >
    test
  </label>
  <input
    autocomplete="off"
    class="c2"
  />
</div>
`;
